native/src/commonOsmAndCore.h

```diff
@@ -352,6 +352,44 @@ class MapDataObject {
 	}
 };
 
+struct RenderableObject {
+	MapDataObject* obj;
+
+	// general
+	int64_t id;
+	std::string type;
+	std::vector<std::pair<int, int>> points;
+	std::vector<tag_value> types;
+	std::vector<tag_value> additionalTypes;
+	bool visible = false;
+
+	// points
+	std::string mainIcon;
+	std::vector<std::string> additionalIcons;
+	std::string shield;
+	double iconX;
+	double iconY;
+	int iconOrder;
+	int iconSize;
+
+	// text
+	std::string text;
+	int textSize;
+	int textOnPath;
+	
+	RenderableObject(MapDataObject* _obj) : obj(_obj) {
+		if (obj) {
+			id = obj->id;
```

Comment:
Why need to copy points, types and additionalTypes if you can just create methods getPoints(), getTypes() etc. ?